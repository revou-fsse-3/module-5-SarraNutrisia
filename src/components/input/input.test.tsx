import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Input from '.';

describe('Input component', () => {
  it('renders with provided props', () => {
    const placeholder = 'Enter text';
    const { getByPlaceholderText } = render(
      <Input type="text" placeholder={placeholder} />
    );

    const inputElement = getByPlaceholderText(placeholder);
    expect(inputElement).toMatchSnapshot();
  });

  it('calls onChange callback when input value changes', () => {
    const onChangeMock = jest.fn();
    const { getByPlaceholderText } = render(
      <Input type="text" placeholder="Enter text" onChange={onChangeMock} />
    );

    const inputElement = getByPlaceholderText('Enter text');
    fireEvent.change(inputElement, { target: { value: 'Hello' } });

    expect(onChangeMock).toHaveBeenCalled();

    expect(inputElement).toMatchSnapshot();
  });

  it('renders with additional className', () => {
    const placeholder = 'Enter text';
    const additionalClass = 'custom-class';
    const { getByPlaceholderText } = render(
      <Input
        type="text"
        placeholder={placeholder}
        className={additionalClass}
      />
    );

    const inputElement = getByPlaceholderText(placeholder);
    expect(inputElement).toHaveClass(additionalClass);
  });
});
