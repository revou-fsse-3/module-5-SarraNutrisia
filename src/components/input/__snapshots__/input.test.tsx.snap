// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component calls onChange callback when input value changes 1`] = `
<input
  placeholder="Enter text"
  type="text"
  value=""
/>
`;

exports[`Input component renders with provided props 1`] = `
<input
  placeholder="Enter text"
  type="text"
  value=""
/>
`;
